# Exploring dataset



In [11]:
df = read.csv("ool_pds.csv")

In [12]:
dim(df)

[1] 2294  436

* There are 2,294 observations and 436 variables.

## Case ID

* The first variable is the case id. It is a sequential integer. 
* No missing value.

In [29]:
summary(df['CASEID'])
str(df['CASEID'])
sum(is.na(df['CASEID']))
c(min(df['CASEID']), max(df['CASEID']))

     CASEID      
 Min.   :   1.0  
 1st Qu.: 574.2  
 Median :1147.5  
 Mean   :1147.5  
 3rd Qu.:1720.8  
 Max.   :2294.0  

'data.frame':	2294 obs. of  1 variable:
 $ CASEID: int  1 2 3 4 5 6 7 8 9 10 ...


[1] 0

[1]    1 2294

## Case ID (Wave 1)

* Wave 1 case id is a sequential integer.
* All observations have wave 1 case id.
* Case ID are not sequential

In [27]:
str(df['W1_CASEID'])
sum(is.na(df['W1_CASEID']))
c(min(df['W1_CASEID']), max(df['W1_CASEID']))

'data.frame':	2294 obs. of  1 variable:
 $ W1_CASEID: int  2 3 4 5 6 7 8 9 10 11 ...


[1] 0

[1]    2 2426

## Case ID (Wave 2)

* Wave 2 case id is also an integer.
* There are 693 missing values.

In [24]:
str(df['W2_CASEID2'])
sum(is.na(df['W2_CASEID2']))

'data.frame':	2294 obs. of  1 variable:
 $ W2_CASEID2: int  820 NA 1198 240 NA 350 926 NA 845 15 ...


[1] 693

### Wave 1

* There are 276 variables (out of total of 436 variables) are related to Wave 1.

In [35]:
w1_vars = names(df)[grepl('^W1_', names(df))]
length(w1_vars)
w1_vars

[1] 276

[1] "W1_CASEID"         "W1_TM_START"       "W1_TM_FINISH"     
  [4] "W1_WEIGHT1"        "W1_WEIGHT2"        "W1_WEIGHT3"       
  [7] "W1_A1"             "W1_A2"             "W1_A4"            
 [10] "W1_A5A"            "W1_A5A_OTHER_TEXT" "W1_A5B"           
 [13] "W1_A5B_OTHER_TEXT" "W1_A10"            "W1_A11"           
 [16] "W1_A12"            "W1_A12A"           "W1_B1"            
 [19] "W1_B2"             "W1_B3"             "W1_B4"            
 [22] "W1_C1"             "W1_C1_OTHER_TEXT"  "W1_C1B"           
 [25] "W1_C1C"            "W1_C2"             "W1_D1"            
 [28] "W1_D2"             "W1_D3"             "W1_D4"            
 [31] "W1_D5"             "W1_D6"             "W1_D7"            
 [34] "W1_D8"             "W1_D9"             "W1_D10"           
 [37] "W1_D11"            "W1_D12"            "W1_D13"           
 [40] "W1_D14"            "W1_D15"            "W1_D16"           
 [43] "W1_D17"            "W1_D18"            "W1_D19"           
 [46] "W1_D20"            "W1_E1"             "W1_E2"            
 [49] "W1_E2A"            "W1_E3"             "W1_E4"            
 [52] "W1_E5_1"           "W1_E5_2"           "W1_E5_3"          
 [55] "W1_E5_4"           "W1_E5_5"           "W1_E5_6"          
 [58] "W1_E5_7"           "W1_E61_A"          "W1_E61_B"         
 [61] "W1_E61_C"          "W1_E61_D"          "W1_E62_A"         
 [64] "W1_E62_B"          "W1_E62_C"          "W1_E62_D"         
 [67] "W1_E63_A"          "W1_E63_B"          "W1_E63_C"         
 [70] "W1_E63_D"          "W1_E64_A"          "W1_E64_B"         
 [73] "W1_E64_C"          "W1_E64_D"          "W1_E7"            
 [76] "W1_E8"             "W1_E9_A"           "W1_E9_B"          
 [79] "W1_E9_C"           "W1_E9_D"           "W1_E9_REFUSED"    
 [82] "W1_F1"             "W1_F1A"            "W1_F2"            
 [85] "W1_F2A"            "W1_F3"             "W1_F4_A"          
 [88] "W1_F4_B"           "W1_F4_C"           "W1_F4_D"          
 [91] "W1_F4_E"           "W1_F5_A"           "W1_F5_B"          
 [94] "W1_F5_C"           "W1_F5_D"           "W1_F5_E"          
 [97] "W1_F6"             "W1_G2"             "W1_G2B"           
[100] "W1_G2C"            "W1_G3A"            "W1_G3B"           
[103] "W1_G3C"            "W1_H1"             "W1_H2"            
[106] "W1_H3"             "W1_H4"             "W1_H5"            
[109] "W1_H6"             "W1_H7"             "W1_H8"            
[112] "W1_I1"             "W1_I2"             "W1_J1_A"          
[115] "W1_J1_B"           "W1_J1_C"           "W1_J1_D"          
[118] "W1_J1_E"           "W1_J1_F"           "W1_J1_G"          
[121] "W1_J2"             "W1_J3A_A"          "W1_J3A_B"         
[124] "W1_J3A_C"          "W1_K1_A"           "W1_K1_B"          
[127] "W1_K1_C"           "W1_K1_D"           "W1_K4"            
[130] "W1_K5"             "W1_L1_A"           "W1_L1_B"          
[133] "W1_L1_C"           "W1_L2_1"           "W1_L2_2"          
[136] "W1_L2_3"           "W1_L2_4"           "W1_L2_5"          
[139] "W1_L2_REFUSED"     "W1_L3"             "W1_L4_A"          
[142] "W1_L4_B"           "W1_L4_C"           "W1_L4_D"          
[145] "W1_L5_A"           "W1_L5_B"           "W1_L5_C"          
[148] "W1_L5_D"           "W1_L5_E"           "W1_L5_F"          
[151] "W1_M1"             "W1_M1_OTHER_TEXT"  "W1_M1A"           
[154] "W1_M5"             "W1_M2"             "W1_M3"            
[157] "W1_M4"             "W1_M6"             "W1_M7"            
[160] "W1_M8"             "W1_M9"             "W1_M10"           
[163] "W1_M11"            "W1_M12"            "W1_N1A"           
[166] "W1_N1B"            "W1_N1C"            "W1_N1D"           
[169] "W1_N1E"            "W1_N1F"            "W1_N1G"           
[172] "W1_N1H"            "W1_N1I"            "W1_N1J"           
[175] "W1_N1K"            "W1_N1L"            "W1_N1M"           
[178] "W1_N1N"            "W1_O1"             "W1_O3"            
[181] "W1_O4"             "W1_O5"         

[1] 436

### Wave 2

* There are 135 variables (out of total of 436 variables) are related to Wave 2.

In [37]:
w2_vars = names(df)[grepl('^W2_', names(df))]
length(w2_vars)
w2_vars

[1] 135

[1] "W2_CASEID2"       "W2_TM_START"      "W2_TM_FINISH"    
  [4] "W2_DURATION"      "W2_QFLAG"         "W2_WEIGHT1"      
  [7] "W2_WEIGHT2"       "W2_WEIGHT3"       "W2_QB1A"         
 [10] "W2_QB1B"          "W2_QB1C"          "W2_QB1D"         
 [13] "W2_QB1D_TEXT"     "W2_QB2A"          "W2_QB2B"         
 [16] "W2_QB3"           "W2_QE1"           "W2_QE1A"         
 [19] "W2_QE2"           "W2_QE2A"          "W2_QE3"          
 [22] "W2_QF1A"          "W2_QF1B"          "W2_QF1C"         
 [25] "W2_QF1D"          "W2_QF2A"          "W2_QF2B"         
 [28] "W2_QF2C"          "W2_QF2D"          "W2_QF3A"         
 [31] "W2_QF3B"          "W2_QF3C"          "W2_QF3D"         
 [34] "W2_QF4A"          "W2_QF4B"          "W2_QF4C"         
 [37] "W2_QF4D"          "W2_QF5A"          "W2_QF5B"         
 [40] "W2_QF5C"          "W2_QF5D"          "W2_QF6A"         
 [43] "W2_QF6B"          "W2_QF6C"          "W2_QF6D"         
 [46] "W2_QF7A"          "W2_QF7B"          "W2_QF7C"         
 [49] "W2_QF7D"          "W2_QF8"           "W2_QF9"          
 [52] "W2_QF10A"         "W2_QF10B"         "W2_QF10C"        
 [55] "W2_QF10D"         "W2_QG_1"          "W2_QG_2"         
 [58] "W2_QG_3"          "W2_QH1"           "W2_QH2"          
 [61] "W2_QH2A"          "W2_QI1"           "W2_QI2"          
 [64] "W2_QI3"           "W2_QI4_A"         "W2_QI4_B"        
 [67] "W2_QI4_C"         "W2_QI4_D"         "W2_QDISP11"      
 [70] "W2_J1_A"          "W2_J1_B"          "W2_J1_C"         
 [73] "W2_J1_D"          "W2_QK1"           "W2_QK2"          
 [76] "W2_QK3"           "W2_QK3A"          "W2_QL1A"         
 [79] "W2_QL1B"          "W2_QL2A"          "W2_QL2B"         
 [82] "W2_QL2C"          "W2_QL3"           "W2_QL4"          
 [85] "W2_QM8"           "W2_QM9"           "W2_QM10"         
 [88] "W2_QM11"          "W2_QM12"          "W2_QN1"          
 [91] "W2_QN2"           "W2_QN3"           "W2_QN4"          
 [94] "W2_QO1"           "W2_QO2"           "W2_QO3"          
 [97] "W2_ORDER_QB3_1"   "W2_ORDER_QB3_2"   "W2_ORDER_QB3_3"  
[100] "W2_ORDER_QB3_4"   "W2_ORDER_QE1A"    "W2_ORDER_QE2A"   
[103] "W2_ORDER_QF1_A"   "W2_ORDER_QF1_B"   "W2_ORDER_QF1_C"  
[106] "W2_ORDER_QF1_D"   "W2_ORDER_QF2_A"   "W2_ORDER_QF2_B"  
[109] "W2_ORDER_QF2_C"   "W2_ORDER_QF2_D"   "W2_ORDER_QF3_A"  
[112] "W2_ORDER_QF3_B"   "W2_ORDER_QF3_C"   "W2_ORDER_QF3_D"  
[115] "W2_ORDER_QF4_A"   "W2_ORDER_QF4_B"   "W2_ORDER_QF4_C"  
[118] "W2_ORDER_QF4_D"   "W2_ORDER_QF5_A"   "W2_ORDER_QF5_B"  
[121] "W2_ORDER_QF5_C"   "W2_ORDER_QF5_D"   "W2_ORDER_QF6_A"  
[124] "W2_ORDER_QF6_B"   "W2_ORDER_QF6_C"   "W2_ORDER_QF6_D"  
[127] "W2_ORDER_QF7_A"   "W2_ORDER_QF7_B"   "W2_ORDER_QF7_C"  
[130] "W2_ORDER_QF7_D"   "W2_ORDER_QF10_A"  "W2_ORDER_QF10_B" 
[133] "W2_ORDER_QF10_C"  "W2_ORDER_QF10_D"  "W2_ORDER_M10_M11"

In [38]:
summary(df['W1_A5A'])

     W1_A5A      
 Min.   :-1.000  
 1st Qu.: 2.000  
 Median : 2.000  
 Mean   : 1.746  
 3rd Qu.: 2.000  
 Max.   : 3.000  
 NA's   :511     

In [39]:
hist(df['W1_A5A'])

ERROR: Error in hist.default(df["W1_A5A"]): 'x' must be numeric


In [58]:
library(gmodels)
x = factor(df$PPETHM, labels = c("White", "Black", "Non-Hispanic", "Hispanic", "Multi-race"))
str(x)

y = factor(df$W1_A5A, labels = c("Refused", "John McCain", "Barack Obama", "Other"))
str(y)
CrossTable(x, y, dnn = c("Race", "Candidate")) 
CrossTable(df$PPETHM, df$W1_A5A, dnn = c("Race", "Candidate")) 

 Factor w/ 5 levels "White","Black",..: 4 2 2 2 1 1 2 1 1 1 ...
 Factor w/ 4 levels "Refused","John McCain",..: 2 NA 3 3 3 NA 3 NA 3 2 ...

 
   Cell Contents
|-------------------------|
|                       N |
| Chi-square contribution |
|           N / Row Total |
|           N / Col Total |
|         N / Table Total |
|-------------------------|

 
Total Observations in Table:  1783 

 
             | Candidate 
        Race |      Refused |  John McCain | Barack Obama |        Other |    Row Total | 
-------------|--------------|--------------|--------------|--------------|--------------|
       White |           10 |          332 |          278 |           14 |          634 | 
             |        0.042 |      271.193 |       82.208 |        2.446 |              | 
             |        0.016 |        0.524 |        0.438 |        0.022 |        0.356 | 
             |        0.333 |        0.853 |        0.208 |        0.538 |              | 
             |        0.006 |   

In [63]:
x = factor(df$PPINCIMP)
str(x)

y = factor(df$W1_A5A, labels = c("Refused", "John McCain", "Barack Obama", "Other"))
str(y)
CrossTable(x, y, dnn = c("Household Income", "Candidate")) 


 Factor w/ 19 levels "1","2","3","4",..: 12 7 15 6 15 13 15 2 10 16 ...
 Factor w/ 4 levels "Refused","John McCain",..: 2 NA 3 3 3 NA 3 NA 3 2 ...

 
   Cell Contents
|-------------------------|
|                       N |
| Chi-square contribution |
|           N / Row Total |
|           N / Col Total |
|         N / Table Total |
|-------------------------|

 
Total Observations in Table:  1783 

 
                 | Candidate 
Household Income |      Refused |  John McCain | Barack Obama |        Other |    Row Total | 
-----------------|--------------|--------------|--------------|--------------|--------------|
               1 |            2 |            4 |           68 |            0 |           74 | 
                 |        0.458 |        9.136 |        2.800 |        1.079 |              | 
                 |        0.027 |        0.054 |        0.919 |        0.000 |        0.042 | 
                 |        0.067 |        0.010 |        0.051 |        0.000 |             

In [68]:
x = factor(df$PPEDUC)
str(x)
y = factor(df$W1_A5A, labels = c("Refused", "John McCain", "Barack Obama", "Other"))
str(y)
CrossTable(df$PPEDUCAT, df$W1_A5A, dnn = c("Education", "Candidate")) 

 Factor w/ 14 levels "1","2","3","4",..: 9 7 12 13 12 7 12 9 12 10 ...
 Factor w/ 4 levels "Refused","John McCain",..: 2 NA 3 3 3 NA 3 NA 3 2 ...

 
   Cell Contents
|-------------------------|
|                       N |
| Chi-square contribution |
|           N / Row Total |
|           N / Col Total |
|         N / Table Total |
|-------------------------|

 
Total Observations in Table:  1783 

 
             | Candidate 
   Education |        -1 |         1 |         2 |         3 | Row Total | 
-------------|-----------|-----------|-----------|-----------|-----------|
           1 |         4 |        23 |       101 |         1 |       129 | 
             |     1.542 |     0.940 |     0.182 |     0.413 |           | 
             |     0.031 |     0.178 |     0.783 |     0.008 |     0.072 | 
             |     0.133 |     0.059 |     0.075 |     0.038 |           | 
             |     0.002 |     0.013 |     0.057 |     0.001 |           | 
-------------|-----------|-----------|-

In [67]:
summary(df$PPEDUCAT)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   2.000   3.000   2.806   4.000   4.000 

In [69]:
CrossTable(df$PPRENT, df$W1_A5A, dnn = c("Home Ownership", "Candidate")) 


 
   Cell Contents
|-------------------------|
|                       N |
| Chi-square contribution |
|           N / Row Total |
|           N / Col Total |
|         N / Table Total |
|-------------------------|

 
Total Observations in Table:  1783 

 
               | Candidate 
Home Ownership |        -1 |         1 |         2 |         3 | Row Total | 
---------------|-----------|-----------|-----------|-----------|-----------|
             1 |        21 |       337 |       832 |        18 |      1208 | 
               |     0.022 |    20.469 |     6.124 |     0.008 |           | 
               |     0.017 |     0.279 |     0.689 |     0.015 |     0.678 | 
               |     0.700 |     0.866 |     0.622 |     0.692 |           | 
               |     0.012 |     0.189 |     0.467 |     0.010 |           | 
---------------|-----------|-----------|-----------|-----------|-----------|
             2 |         9 |        41 |       483 |         8 |       541 | 
              